mathics/builtin/numbers/calculus.py

```diff
@@ -772,11 +772,15 @@ class FindMaximum(_BaseFinder):
     >> Quiet[FindMaximum[-phi[x] + x, {x, 1.2}, Method->"Newton"]]
      = {0.5, {x -> 1.00001}}
     >> Clear[phi];
-    For a not so well behaving function, the result can be less accurate:
-    >> FindMaximum[-Exp[-1/x^2]+1., {x,1.2}, MaxIterations->10]
-     : The maximum number of iterations was exceeded. The result might be inaccurate.
-     = FindMaximum[-Exp[-1 / x ^ 2] + 1., {x, 1.2}, MaxIterations -> 10]
+
     """
+    #  WMA does not have a problem here. So this is something in our implementation
+    # Also, we can get either a MaxIterations failure, *or* a $IterationLimit error.
+    #
+    # For a not so well behaving function, the result can be less accurate:
+    # >> FindMaximum[-Exp[-1/x^2]+1., {x,1.2}, MaxIterations->10]
```

Comment:
```
In[4]:= FindMaximum[-Exp[-1/x^2]+1., {x,1.2}, MaxIterations->2]                 

FindMaximum::cvmit: 
   Failed to converge to the requested accuracy or precision within 2
     iterations.

Out[4]= {1., {x -> 0.168461}}
```

Reply:
Ok. For FindMaximum, at iterations 2 it fails. At 10 though it doesn't which is what the test was. 

Also, when it fails, it gives a value, and Mathics3 does not.